nixos/modules/services/networking/knot.nix

```diff
@@ -98,17 +98,52 @@ in {
         Type = "notify";
         ExecStart = "${cfg.package}/bin/knotd --config=${configFile} --socket=${socketFile} ${concatStringsSep " " cfg.extraArgs}";
         ExecReload = "${knot-cli-wrappers}/bin/knotc reload";
-        CapabilityBoundingSet = "CAP_NET_BIND_SERVICE CAP_SETPCAP";
-        AmbientCapabilities = "CAP_NET_BIND_SERVICE CAP_SETPCAP";
-        NoNewPrivileges = true;
         User = "knot";
+        Group = "knot";
```

Comment:
The group was missing, which means it defaulted to root ever since 88029bce39bd485fc07f1b2aa111c3ee9d12e684 :disappointed:

Reply:
I filed #148555 as a followup, this sounds like something that could have been prevented.